riscv/src/compiler.rs

```diff
@@ -35,6 +35,17 @@ pub fn compile_riscv_asm(mut assemblies: BTreeMap<String, String>) -> String {
     replace_dynamic_label_references(&mut statements, &objects);
 
     // Sort the objects according to the order of the names in object_order.
+    // With the single exception: If there is large object, put that at the end.
+    // The idea behind this is that there might be a single gigantic object representing the heap
+    // and putting that at the end should keep memory addresses small.
+    let mut large_objects = objects
+        .iter()
+        .filter(|(_name, data)| data.iter().map(|d| d.size()).sum::<usize>() > 0x2000);
+    if let (Some((heap, _)), None) = (large_objects.next(), large_objects.next()) {
```

Comment:
If there are more than 1 big object, then none are moved to the end? Wouldn't it be best to move the largest one?

Reply:
0x2000 is already too large. I mean ok, if there are no gaps in the access that it could still be fine. Let's see if this helps. We will get a proper error message now anyway.